cli/translator.js

```diff
@@ -0,0 +1,58 @@
+import fs from 'fs/promises'
+
+export default async function translator(path) {
+  const cwd = process.cwd()
+  const rootPath = await searchRootFolder(cwd)
+  if (!rootPath) {
+    throw Error(`Cannot translate using ${cwd} folder`)
+  }
+
+  const relativeRoot = removeAppFromPath(cwd.substr(rootPath.length))
+  const isAppPath = path?.startsWith(`app`)
+  const resolvePath = isAppPath 
+    ? removeAppFromPath(path) 
+    : path ?? ``
+
+  const translatedPath = isAppPath
+    ? resolvePath
+    : `${relativeRoot ? relativeRoot + `/` : ``}` + resolvePath
+
+  return translatedPath
+}
+
+// translate app/a/b/c => a/b/c
+function removeAppFromPath(path) {
+  return path
+    .split(`/`)
+    .filter(dir => dir.length > 0 && dir !== `app`)
+    .join(`/`)
+}
+
+async function searchRootFolder(cwd) {
+  let rootPath = ``
+  const folders = cwd.split(`/`)
+
+  if (folders.length > 0) {
+    const promises = []
+    const paths = []
+    let path = ``
+
+    for (let i = 0; i < folders.length; i++) {
+      const childPath = i === folders.length - 1 ? folders[i] : folders[i] + `/`
+      path += childPath
+      paths.push(path)
+      promises.push(fs.readdir(path))
+    }
+
+    const foldersInfo = await Promise.all(promises)
+
+    for (let i = 0; i < foldersInfo.length; i++) {
+      const maybeRoot = foldersInfo[i].indexOf(`manifest.json`) !== -1
+      if (maybeRoot) {
+        rootPath = paths[i]
+      }
+    }
+  }
+
+  return rootPath
```

Comment:
This is good. Just for fun, I thought I'd share a shorter way of doing it :)

```javascript
import {stat} from 'fs/promises'
import {resolve} from 'path'

async function searchRootFolder(path) {
  const hasManifest = !!(await stat(`${path}/manifest.json`).catch(e => false))
  if (hasManifest) {
    return path
  } else {
    return await searchRootFolder(resolve([path, `..`]))
  }
}
```

Reply:
I think about a little recursive also, the reason i don't want to crawl folder by folder because of waiting time. 
- Recursive: If our folder have 10 levels and await take 100ms each => we gonna wait for 2s
- Promise.all: If our folder have 10 levels and await take 100ms each => we gonna wait for 100ms due to Promise.all

Reply:
I'm not sure about the performance for listing all the files in a directory using `readdir` (your code), but if using `stat` it is insanely fast. You can do several of these in less than a millisecond :)

Also, it would be quite insane to have more than 3 namespaces deep. 10 wouldn't make sense.

Also, keep in mind that we're not interested in most of the folders. Perhaps the path is `/home/mika/myProjects/work/whenBored/andHaveTime/tiden/app/myNamespace` which is 9 folders, however only `myNamespace`, `app` and `tiden` actually need to be checked to find the root.

Reply:
totally agree that `stat` is really insanely fast, on it

Reply:
DONE